plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -258,15 +265,40 @@ instance Binary GetHlintSettings
 
 type instance RuleResult GetHlintSettings = (ParseFlags, [Classify], Hint)
 
-getHlintSettingsRule :: HlintUsage -> Rules ()
+getHlintSettingsRule :: HlintUsage -> Action (ParseFlags, [Classify], Hint)
 getHlintSettingsRule usage =
-    defineNoFile $ \GetHlintSettings ->
       liftIO $ case usage of
           HlintEnabled cmdArgs -> argsSettings cmdArgs
           HlintDisabled        -> fail "hlint configuration unspecified"
 
 -- ---------------------------------------------------------------------
 
+newtype Config = Config [String]
+
+properties :: Properties '[ 'PropertyKey "flags" ('TArray String)]
```

Comment:
Please add `properties` to plugin descriptor, for example:
https://github.com/haskell/haskell-language-server/blob/2683a92af0ce043254d7ba8b4d379d8e7d9075fb/ghcide/src/Development/IDE/Plugin/TypeLenses.hs#L93